bindings/rabbitmq/rabbitmq.go

```diff
@@ -84,34 +97,81 @@ func (r *RabbitMQ) Init(_ context.Context, metadata bindings.Metadata) error {
 		return err
 	}
 
-	conn, err := amqp.Dial(r.metadata.Host)
+	err = r.connect()
 	if err != nil {
 		return err
 	}
+	r.notifyRabbitChannelClose = make(chan *amqp.Error, 1)
+	r.channel.NotifyClose(r.notifyRabbitChannelClose)
```

Comment:
NotifyClose returns an Error struct(https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Error), which also has a Recover attribute, which returns `true when this error can be recovered by retrying later or with different parameters`. Should we check Error.Retry's value before retrying to connect? And, if we still want to re-connect, should we at least wait for sometime, so that we don't keep retrying a connection which may be un-retriable at the moment.

Reply:
Yes, you are right! If the recover attribute is false, we should wait for sometime. 
Since this is only notified when the channel is closed, and the error struct is a common struct, so I'd more like to keep the reconnect part.